apps/server/src/shared/repo/dashboard/dashboard.model.mapper.spec.ts

```diff
@@ -92,27 +97,45 @@ describe('dashboard model mapper', () => {
 
 		it('should detect changes to gridElement Collection', async () => {
 			const user = userFactory.build();
+			await em.persistAndFlush(user);
 			const dashboardId = new ObjectId().toString();
 			const elementId = new ObjectId().toString();
 			const oldElementId = new ObjectId().toString();
 			const newElementId = new ObjectId().toString();
-			const originalDashboard = new DashboardModelEntity({ id: dashboardId, user: new ObjectId().toString() });
-			originalDashboard.gridElements.add(new DashboardGridElementModel(oldElementId));
-			originalDashboard.gridElements.add(new DashboardGridElementModel(new ObjectId().toString()));
+			// TODO: use builder
+			const originalDashboard = new DashboardModelEntity({ id: dashboardId, user });
+			originalDashboard.gridElements.add(
+				new DashboardGridElementModel({
+					id: oldElementId,
+					xPos: 1,
+					yPos: 1,
+					references: [courseFactory.build({ students: [user] })],
+					dashboard: originalDashboard,
+				})
+			);
+			originalDashboard.gridElements.add(
+				new DashboardGridElementModel({
+					id: elementId,
+					xPos: 1,
+					yPos: 2,
+					references: [courseFactory.build({ students: [user] })],
+					dashboard: originalDashboard,
+				})
+			);
 			await em.persistAndFlush(originalDashboard);
 
 			const dashboard = new DashboardEntity(dashboardId, {
 				grid: [
 					{
-						pos: { x: 1, y: 2 },
+						pos: { x: 2, y: 1 },
 						gridElement: GridElement.FromPersistedReference(elementId, courseFactory.build({ students: [user] })),
 					},
 					{
-						pos: { x: 1, y: 4 },
+						pos: { x: 2, y: 2 },
 						gridElement: GridElement.FromPersistedReference(newElementId, courseFactory.build({ students: [user] })),
 					},
 				],
-				userId: new ObjectId().toString(),
+				userId: user.id,
```

Comment:
dashboard if it's an entity should know a user, not a userid
user might not need to be flushed( persistandflush) while simple registration in the em might be enough. @uidp ?